scripts/spack/packages/axom/package.py

```diff
@@ -95,7 +95,7 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("conduit~hdf5", when="~hdf5")
 
     # HDF5 needs to be the same as Conduit's
-    depends_on("hdf5@1.8.19:1.8.999~cxx~shared~fortran", when="+hdf5")
+    depends_on("hdf5@1.8.19:1.8.999~cxx~fortran", when="+hdf5")
```

Comment:
Minor: should the `~cxx` and `~fortran` restrictions on `hdf5` be moved to our `packages.yaml` ?

I.e. we do not expect problems building `axom` against an hdf5@1.8 that was built w/ `+cxx` or `+fortran`, we just choose not to build these internally for `axom`, right?

(Inspired by a relatively recent conversation w/ @scheibelp )

Reply:
I think with the new concretizer we dont needs this restriction in axom and can just leave it up to conduit but the clingo concretizer isn't default yet.